### PR TITLE
feat:[CDS-81011]: fix for capability check for Github App connector for CloudFormation, Azure and Terraform

### PR DESCRIPTION
**Pull request by sourabh-awashti**
## Describe your changes
Changes are regarding the fix in capability check for delegate tasks for CloudFormation, Azure and Terraform and Terragrunt swimlanes. We are converting account level git connector to repo level git connector before sending it to delegate tasks or capability check.
Tests Done:
1. Validation with Repo level Github connector for all of the above mentioned swimlanes.
2. Validation with Account level Github connector for all mentioned swimlanes and fetch template call in cloudformation.
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54090)

### CODE REVIEW

**Review by vlica-harness — APPROVED** [review]

**Comment by sourabh-awashti** [comment]
trigger uts

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java [hunk 1]
```diff
@@ -1261,9 +1262,20 @@ public List<TaskSelector> getDelegateSelectors(ConnectorInfoDTO connectorInfoDTO
...
+  public ScmConnector getScmConnector(
+      ScmConnector scmConnector, String accountIdentifier, GitConfigDTO gitConfigDTO, GitStoreConfig gitStoreConfig) {
     if (scmConnector instanceof GithubConnectorDTO && isGithubAppAuth((GithubConnectorDTO) scmConnector)
         && cdFeatureFlagHelper.isEnabled(accountIdentifier, CDS_GITHUB_APP_AUTHENTICATION)) {
+      if (gitStoreConfig == null) {
+        GithubConnectorDTO githubConnectorDTO = (GithubConnectorDTO) scmConnector;
+        GithubStore githubStore =
+            GithubStore.builder()
+                .repoName(ParameterField.<String>builder().value(githubConnectorDTO.getValidationRepo()).build())
+                .build();
```
Comment (yogesh-chauhan) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java:1274]: we can create a new method that takes reponame as the 3rd argument. That will not change existing callers

**Review by yogesh-chauhan — CHANGES_REQUESTED** [review]

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java (hunk 1 above)
Comment (sourabh-awashti) [reply 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java]: done

**Review by yogesh-chauhan — DISMISSED** [review]

**Review by vlica-harness — APPROVED** [review]

**Comment by sourabh-awashti** [comment]
trigger uts

**Comment by sourabh-awashti** [comment]
trigger uts

**Comment by sourabh-awashti** [comment]
trigger messagecheck

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 5 of 12 files at r1, 5 of 7 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54090)* status: Some of the checks are too old. 12 file(s) to review. 1 of 2 approvals obtained.

**Review by senjucanon2 — APPROVED** [review]
> Reviewed 5 of 12 files at r1, 5 of 7 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54090)* status: Some of the checks are too old.

**Comment by sourabh-awashti** [comment]
trigger review

**Pull request merged**